apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts

```diff
@@ -0,0 +1,566 @@
+          Cancel
+        </button>
+        <button type="submit" class="btn btn-primary">Submit</button>
+      </div>
+    </form>
+
+    <script>
+      $(function () {
+        const validateCourse = function () {
+          let element = $('select[name="to_course_id"]');
+          let elementDOM = element.get(0);
+
+          elementDOM.setCustomValidity('');
+          if (elementDOM.validity.valueMissing) {
+            $('#invalidIdMessage').text('Please choose a course');
+          } else {
+            $('#invalidIdMessage').text('');
+          }
+        };
+
+        $('input[name="to_course_id"]').on('input', validateCourse);
+        $('input[name="to_course_id"]').on('change', validateCourse);
+
+        $('form[name="copy-question-form"]').submit(function (event) {
+          validateCourse();
+          if ($(this).get(0).checkValidity() === false) {
+            event.preventDefault();
+            event.stopPropagation();
+          }
+          $(this).addClass('was-validated');
+        });
+      });
+    </script>
+  `;
+}
+
+function PubliclyShareModal({ csrfToken, qid }: { csrfToken: string; qid: string }) {
+  return Modal({
+    id: 'publiclyShareModal',
+    title: 'Confirm Publicly Share Question',
+    body: html`
+      <p>Are you sure you want to publicly share this question?</p>
+      <p>
+        Once this question is publicly shared, anyone will be able to view it or use it as a part of
+        their course. This operation cannot be undone.
+      </p>
+      ${isEnterprise()
+        ? html`
+            <p>
+              You retain full ownership of all shared content as described in the
+              <a href="https://www.prairielearn.com/legal/terms#2-user-content" target="_blank"
+                >Terms of Service</a
+              >. To allow PrairieLearn to share your content to other users you agree to the
+              <a
+                href="https://www.prairielearn.com/legal/terms#3-user-content-license-grant"
+                target="_blank"
+                >User Content License Grant</a
+              >.
+            </p>
+          `
+        : ''}
+    `,
+    footer: html`
+      <input type="hidden" name="__action" value="share_publicly" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+      <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+      <button class="btn btn-primary" type="submit">Publicly Share "${qid}"</button>
+    `,
+  });
+}
+
+function DeleteQuestionModal({
+  qid,
+  assessmentsWithQuestion,
+  csrfToken,
+}: {
+  qid: string;
+  assessmentsWithQuestion: SelectedAssessments[];
+  csrfToken: string;
+}) {
+  return Modal({
+    id: 'delete-question-form',
+    title: 'Delete question',
+    body: html`
+      <p>
+        Are you sure you want to delete the question
+        <strong>${qid}</strong>?
+      </p>
+      ${assessmentsWithQuestion
```

Comment:
An empty array is still truthy in JavaScript. You want this instead:

```suggestion
      ${assessmentsWithQuestion.length
```